src/sema/expression.rs

```diff
@@ -1314,6 +1310,143 @@ fn coerce_number(
     })
 }
 
+fn number_literal(
+    loc: &pt::Loc,
+    base: &str,
+    exp: &str,
+    ns: &Namespace,
+    factor: &BigInt,
+    diagnostics: &mut Vec<Diagnostic>,
+    resolve_to: ResolveTo,
+) -> Result<Expression, ()> {
+    let base = BigInt::from_str(base).unwrap();
```

Comment:
In scientific notation, the correct nomenclature is `coefficient X base^(exponent)`. I think this is the coefficient, right?

Reply:
Oh, I saw that you multiply n by the factor (i.e. coefficient) later.

Reply:
Say `123.456e789`. I've called `123` the base, `456` the fraction, and `789` the exponent. Not sure what the correct nomenclature is.

Reply:
I think the name should be changed. We call base the base of an exponentiation.
For exponentiation, the nomenclature is `coefficient X base^(exponent)`. Coefficient is defined as `integerPart.fractionalPart`.

Reply:
I've given them new names. Let me know what you think

Reply:
I still could figure out what `factor` is.